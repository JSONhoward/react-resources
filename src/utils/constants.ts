import cleverP from '../images/clever-programmer-icon.jpg'
import devEd from '../images/dev-ed-icon.jpg'
import fireship from '../images/fireship-icon.jpg'
import freeCodeCamp from '../images/free-code-camp-icon.jpg'
import netNinja from '../images/net-ninja-icon.jpg'
import webDevSimplified from '../images/web-dev-simplified-icon.jpg'
import rwieruch from '../images/rwieruch-icon.svg'
import sitePoint from '../images/sitepoint.svg'
import smashingMagazine from '../images/smashing-magazine.svg'
import codeCademy from '../images/codecademy.svg'
import udemy from '../images/udemy-icon.svg'
import wesBos from '../images/wes-bos-icon.png'
import medium from '../images/medium-icon.png'
import devTo from '../images/dev-to.svg'

export const webDevChannels: { name: string, id: string, icon: string }[] = [
    {
        id: 'UCFbNIlppjAuEX4znoulh0Cw',
        name: 'Web Dev Simplified',
        icon: webDevSimplified
    },
    {
        id: 'UClb90NQQcskPUGDIXsQEz5Q',
        name: 'Dev Ed',
        icon: devEd
    },
    {
        id: 'UCW5YeuERMmlnqo4oq8vwUpg',
        name: 'Net Ninja',
        icon: netNinja
    },
    {
        id: 'UCsBjURrPoezykLs9EqgamOA',
        name: 'Fireship',
        icon: fireship
    },
    {
        id: 'UC8butISFwT-Wl7EV0hUK0BQ',
        name: 'Free Code Camp',
        icon: freeCodeCamp
    },
    {
        id: 'UCqrILQNl5Ed9Dz6CGMyvMTQ',
        name: 'Clever Programmer',
        icon: cleverP
    }
]

export const mockArray: {id: string, url: string, title: string, description: string, thumb: string}[] = [
    {
        id: 'someId',
        url: 'https://www.youtube.com/watch?v=Faz6jkFTaa0',
        title: 'title goes here',
        description: 'lorem ipsum',
        thumb: `https://via.placeholder.com/300x200`
    },
    {
        id: 'someId',
        url: 'https://www.youtube.com/watch?v=Faz6jkFTaa0',
        title: 'title goes here',
        description: 'lorem ipsum',
        thumb: `https://via.placeholder.com/300x200`
    }, {
        id: 'someId',
        url: 'https://www.youtube.com/watch?v=Faz6jkFTaa0',
        title: 'title goes here',
        description: 'lorem ipsum',
        thumb: `https://via.placeholder.com/300x200`
    }, {
        id: 'someId',
        url: 'https://www.youtube.com/watch?v=Faz6jkFTaa0',
        title: 'title goes here',
        description: 'lorem ipsum',
        thumb: `https://via.placeholder.com/300x200`
    },
    {
        id: 'someId',
        url: 'https://www.youtube.com/watch?v=Faz6jkFTaa0',
        title: 'title goes here',
        description: 'lorem ipsum',
        thumb: `https://via.placeholder.com/300x200`
    }
]

export const onlineCourses: {platform: string, name: string, url: string, icon: string}[] = [
    {
        platform: 'FreeCodeCamp',
        name: 'Learn React in 1 Hour by Building a Movie Search App',
        url: 'https://www.freecodecamp.org/news/learn-react-in-1-hour-by-building-a-movie-search-app/',
        icon: freeCodeCamp
    },
    {
        platform: 'Udemy',
        name: 'React - The Complete Guide (incl Hooks, React Router, Redux)',
        url: 'https://www.udemy.com/course/react-the-complete-guide-incl-redux/',
        icon: udemy
    },
    {
        platform: 'CodeCademy',
        name: 'Learn React.js: Part I/Part II',
        url: 'https://www.codecademy.com/learn/react-101',
        icon: codeCademy
    },
    {
        platform: 'none',
        name: 'Fullstack Advanced React & GraphQL',
        url: 'https://advancedreact.com/',
        icon: wesBos
    }
]

export const onlineBlogs: {name: string, url: string, icon: string}[] = [
    {
        name: 'Medium',
        url: 'https://medium.com/',
        icon: medium
    },
    {
        name: 'Dev.to',
        url: 'https://dev.to/',
        icon: devTo
    },
    {
        name: 'Sitepoint',
        url: 'https://www.sitepoint.com/',
        icon: sitePoint
    },
    {
        name: 'Smashing Magazine',
        url: 'https://www.smashingmagazine.com/',
        icon: smashingMagazine
    },
    {
        name: 'RobinWieruch',
        url: 'https://www.robinwieruch.de/blog',
        icon: rwieruch
    }
]