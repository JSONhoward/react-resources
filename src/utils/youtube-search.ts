const YOUTUBE_SEARCH_URL = (channel: string, limit: string): string => `https://www.googleapis.com/youtube/v3/search?part=snippet&channelId=${channel}${limit}&q=react&type=video&key=${process.env.REACT_APP_YOUTUBE_API}`

export const getVideoList = async (channel: string, limit = 50): Promise<Video[]> => {
    const response = await fetch(YOUTUBE_SEARCH_URL(channel, `&maxResults=${limit.toString()}`))
    const data = await response.json()

    const videos: Video[] = await data['items'].map((result: { id: { videoId: any }; snippet: { title: any; description: any; thumbnails: { high: { url: any } } } }) => {
        return {
            id: result.id.videoId,
            url: `http://www.youtube.com/watch?v=${result.id.videoId}`,
            title: result.snippet.title,
            description: result.snippet.description,
            thumb: result.snippet.thumbnails.high.url
        }
    })

    return videos
}