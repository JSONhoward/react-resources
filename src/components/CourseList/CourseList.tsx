import React from 'react'
import { StyledCourseList } from './CourseList.styles';
import { onlineCourses } from '../../utils/constants';
import Courses from './Courses';

const CourseList = () => {
    const courses = onlineCourses.map((el, i) => {
        return (
            <Courses key={i} icon={el.icon} title={el.name} url={el.url} />
        )
    })

    return (
        <StyledCourseList>
            {courses}
        </StyledCourseList>
    )
}

export default CourseList
