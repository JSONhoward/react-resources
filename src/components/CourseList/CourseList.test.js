import React from 'react'
import {create} from 'react-test-renderer'
import {screen, render} from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import CourseList from './'
import { onlineCourses } from '../../utils/constants'

describe('CourseList Test', () => {
    test('snapshot matches', () => {
        const tree = create(<CourseList />).toJSON()

        expect(tree).toMatchSnapshot()
    })

    test('renders each course as child', () => {
        render(<CourseList />)

        onlineCourses.forEach((course, i) => {
            expect(screen.getByText(course.name)).toBeInTheDOM()
        })
    })
})
