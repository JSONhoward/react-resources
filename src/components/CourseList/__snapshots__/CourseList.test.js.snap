// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CourseList Test snapshot matches 1`] = `
<div
  className="sc-bdnylx bxwOIp"
>
  <div
    className="sc-gtssRu hKLDDf"
  >
    <img
      className="sc-dlnjPT jMGHbI"
      data-testid="courseIcon"
      src="free-code-camp-icon.jpg"
    />
    <p
      className="sc-hKFyIo itAKev"
    >
      Learn React in 1 Hour by Building a Movie Search App
    </p>
    <a
      className="sc-eCApGN cUAoBS"
      data-testid="courseLink"
      href="https://www.freecodecamp.org/news/learn-react-in-1-hour-by-building-a-movie-search-app/"
      rel="noreferrer noopener"
      target="_blank"
    >
      <svg
        fill="currentColor"
        height="2rem"
        size="2rem"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 512 512"
        width="2rem"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M432,320H400a16,16,0,0,0-16,16V448H64V128H208a16,16,0,0,0,16-16V80a16,16,0,0,0-16-16H48A48,48,0,0,0,0,112V464a48,48,0,0,0,48,48H400a48,48,0,0,0,48-48V336A16,16,0,0,0,432,320ZM488,0h-128c-21.37,0-32.05,25.91-17,41l35.73,35.73L135,320.37a24,24,0,0,0,0,34L157.67,377a24,24,0,0,0,34,0L435.28,133.32,471,169c15,15,41,4.5,41-17V24A24,24,0,0,0,488,0Z"
        />
      </svg>
    </a>
  </div>
  <div
    className="sc-gtssRu hKLDDf"
  >
    <img
      className="sc-dlnjPT jMGHbI"
      data-testid="courseIcon"
      src="udemy-icon.svg"
    />
    <p
      className="sc-hKFyIo itAKev"
    >
      React - The Complete Guide (incl Hooks, React Router, Redux)
    </p>
    <a
      className="sc-eCApGN cUAoBS"
      data-testid="courseLink"
      href="https://www.udemy.com/course/react-the-complete-guide-incl-redux/"
      rel="noreferrer noopener"
      target="_blank"
    >
      <svg
        fill="currentColor"
        height="2rem"
        size="2rem"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 512 512"
        width="2rem"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M432,320H400a16,16,0,0,0-16,16V448H64V128H208a16,16,0,0,0,16-16V80a16,16,0,0,0-16-16H48A48,48,0,0,0,0,112V464a48,48,0,0,0,48,48H400a48,48,0,0,0,48-48V336A16,16,0,0,0,432,320ZM488,0h-128c-21.37,0-32.05,25.91-17,41l35.73,35.73L135,320.37a24,24,0,0,0,0,34L157.67,377a24,24,0,0,0,34,0L435.28,133.32,471,169c15,15,41,4.5,41-17V24A24,24,0,0,0,488,0Z"
        />
      </svg>
    </a>
  </div>
  <div
    className="sc-gtssRu hKLDDf"
  >
    <img
      className="sc-dlnjPT jMGHbI"
      data-testid="courseIcon"
      src="codecademy.svg"
    />
    <p
      className="sc-hKFyIo itAKev"
    >
      Learn React.js: Part I/Part II
    </p>
    <a
      className="sc-eCApGN cUAoBS"
      data-testid="courseLink"
      href="https://www.codecademy.com/learn/react-101"
      rel="noreferrer noopener"
      target="_blank"
    >
      <svg
        fill="currentColor"
        height="2rem"
        size="2rem"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 512 512"
        width="2rem"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M432,320H400a16,16,0,0,0-16,16V448H64V128H208a16,16,0,0,0,16-16V80a16,16,0,0,0-16-16H48A48,48,0,0,0,0,112V464a48,48,0,0,0,48,48H400a48,48,0,0,0,48-48V336A16,16,0,0,0,432,320ZM488,0h-128c-21.37,0-32.05,25.91-17,41l35.73,35.73L135,320.37a24,24,0,0,0,0,34L157.67,377a24,24,0,0,0,34,0L435.28,133.32,471,169c15,15,41,4.5,41-17V24A24,24,0,0,0,488,0Z"
        />
      </svg>
    </a>
  </div>
  <div
    className="sc-gtssRu hKLDDf"
  >
    <img
      className="sc-dlnjPT jMGHbI"
      data-testid="courseIcon"
      src="wes-bos-icon.png"
    />
    <p
      className="sc-hKFyIo itAKev"
    >
      Fullstack Advanced React & GraphQL
    </p>
    <a
      className="sc-eCApGN cUAoBS"
      data-testid="courseLink"
      href="https://advancedreact.com/"
      rel="noreferrer noopener"
      target="_blank"
    >
      <svg
        fill="currentColor"
        height="2rem"
        size="2rem"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 512 512"
        width="2rem"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M432,320H400a16,16,0,0,0-16,16V448H64V128H208a16,16,0,0,0,16-16V80a16,16,0,0,0-16-16H48A48,48,0,0,0,0,112V464a48,48,0,0,0,48,48H400a48,48,0,0,0,48-48V336A16,16,0,0,0,432,320ZM488,0h-128c-21.37,0-32.05,25.91-17,41l35.73,35.73L135,320.37a24,24,0,0,0,0,34L157.67,377a24,24,0,0,0,34,0L435.28,133.32,471,169c15,15,41,4.5,41-17V24A24,24,0,0,0,488,0Z"
        />
      </svg>
    </a>
  </div>
</div>
`;
